generator/src/test/java/com/scottlogic/deg/generator/cucumber/utils/InMemoryViolatingOutputTarget.java

```diff
@@ -0,0 +1,25 @@
+package com.scottlogic.deg.generator.cucumber.utils;
+
+import com.google.inject.Inject;
+
+/**
+ * Defines an output target which stores the output data into the test state.
+ */
+public class InMemoryViolatingOutputTarget extends InMemoryOutputTarget {
```

Comment:
I don't think you really need this class, there are not different versions in production code.

Reply:
Instead just make the general InMemoryOutputTarget use the test state to determine returns.

Reply:
as the InMemoryOutputTarget is injected, I am not even sure where to begin spying on it in the tests